api_v3/lib/types/filters/KalturaUserAppRoleFilter.php

```diff
@@ -0,0 +1,167 @@
+		{
+			$totalCount = ($pager->pageIndex - 1) * $pager->pageSize + $resultCount;
+		}
+		else
+		{
+			KalturaFilterPager::detachFromCriteria($c);
+			$totalCount = KuserToUserRolePeer::doCount($c);
+		}
+		
+		KuserToUserRolePeer::setUseCriteriaFilter(true);
+		
+		$response->objects = KalturaUserAppRoleArray::fromDbArray($list, $responseProfile);
+		$response->totalCount = $totalCount;
+		
+		return $response;
+	}
+	
+	/**
+	 * @throws KalturaAPIException
+	 * @throws PropelException
+	 */
+	public function toObject($object_to_fill = null, $props_to_skip = array())
+	{
+		$this->fixCsvFilterProperties();
+		
+		$isAdminSession = kCurrentContext::getCurrentSessionType() === kSessionBase::SESSION_TYPE_ADMIN;
+		
+		// user ks can only retrieve results for himself
+		if (!$isAdminSession)
+		{
+			$kuser = kCurrentContext::getCurrentKsKuser();
+			if (!$kuser)
+			{
+				throw new KalturaAPIException(KalturaErrors::USER_ID_NOT_FOUND, kCurrentContext::$ks_uid);
+			}
+			
+			$this->userIdEqual = $kuser->getId();
+			$this->userIdIn = null;
+		}
+		
+		if ($isAdminSession && !empty($this->userIdEqual))
+		{
+			$kuser = kuserPeer::getKuserByPartnerAndUid(kCurrentContext::getCurrentPartnerId(), $this->userIdEqual);
+			$this->userIdEqual = $kuser ? $kuser->getId() : -1; // set -1 = no result will be returned when the user is missing
+		}
+		
+		if ($isAdminSession && !empty($this->userIdIn))
+		{
+			$this->userIdIn = myKuserUtils::preparePusersToKusersFilter($this->userIdIn);
+		}
+		
+		if (!empty($this->userRoleIdEqual))
+		{
+			$userRole = UserRolePeer::retrieveByPK($this->userRoleIdEqual);
+			$this->userRoleIdEqual = $userRole ? $userRole->getId() : -1;
+		}
+		
+		if (!empty($this->userRoleIdIn))
+		{
+			$this->userRoleIdIn = UserRolePeer::retrieveExistingUserRolesCsvByCsv($this->userRoleIdIn);
+		}
+		
+		if (!empty($this->appGuidEqual))
+		{
+			$appRegistry = KuserToUserRolePeer::getAppGuidById($this->appGuidEqual);
+			$this->appGuidEqual = $appRegistry ? $appRegistry->id : 'null'; // set to 'null' will not return results (if set to '' 'attachToCriteria' will remove it from mysql query)
+		}
+		
+		if(!empty($this->appGuidIn))
+		{
+			$this->appGuidIn = KuserToUserRolePeer::getKsPartnerAppGuidsFromCsv($this->appGuidIn);
+		}
+		
+		return parent::toObject($object_to_fill, $props_to_skip);
+	}
+	
+	public function doFromObject($srcObj, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		/* @var $srcObj AppRoleFilter */
+		parent::doFromObject($srcObj, $responseProfile);
+		
+		if ($srcObj->get('_eq_user_id'))
+		{
+			$this->userIdEqual = $this->prepareKusersToPusersFilter($srcObj->get('_eq_user_id'));
+		}
+		
+		if ($srcObj->get('_in_user_id'))
+		{
+			$this->userIdIn = $this->prepareKusersToPusersFilter($srcObj->get('_in_user_id'));
+		}
+	}
+	
+	protected function fixCsvFilterProperties()
+	{
+		$this->userIdIn = !empty($this->userIdIn) ? kString::csvFixWhitespace($this->userIdIn) : null;
```

Comment:
If you want to trim the space after the comma better do explode and array map with trim